packages/zoe/src/zoe.js

```diff
@@ -631,262 +413,274 @@ function makeZoe(additionalEndowments = {}, vatPowers = {}) {
+            instanceHandle,
+            proposal: cleanedProposal,
+            currentAllocation: arrayToObj(amountsArray, userKeywords),
+            notifier: notifierRec.notifier,
+            updater: notifierRec.updater,
+          };
+          const { innerFacet } = instanceTable.get(instanceHandle);
+          return E(innerFacet)
+            .addOffer(
+              offerHandle,
+              cleanedProposal,
+              offerRecord.currentAllocation,
+            )
+            .then(completeObj => {
+              payoutMap.init(offerHandle, producePromise());
+              offerTable.create(offerRecord, offerHandle);
+              instanceTable.addOffer(instanceHandle, offerHandle);
+              return completeObj;
+            });
+        };
 
-            // if the exitRule.kind is 'waived' the user has no
-            // possibility of completing an offer on demand
-            return harden(offerResult);
+        const makeOfferResult = completeObj => {
+          const offerHook = inviteHandleToOfferHook.get(inviteHandle);
+          const outcomeP = E(offerHook).invokeOfferHook(offerHandle);
+          const offerResult = {
+            offerHandle: HandledPromise.resolve(offerHandle),
+            payout: payoutMap.get(offerHandle).promise,
+            outcome: outcomeP,
+            completeObj,
           };
-          return Promise.all(paymentDepositedPs)
-            .then(recordOffer)
-            .then(makeOfferResult);
-        });
-      },
+          return harden(offerResult);
+        };
+        return Promise.all(paymentDepositedPs)
+          .then(recordOffer)
+          .then(makeOfferResult);
+      });
+    },
 
-      isOfferActive: offerTable.isOfferActive,
-      getOffers: offerHandles =>
-        offerTable.getOffers(offerHandles).map(removeAmountsAndNotifier),
-      getOffer: offerHandle =>
-        removeAmountsAndNotifier(offerTable.get(offerHandle)),
-      getCurrentAllocation: (offerHandle, brandKeywordRecord) =>
-        doGetCurrentAllocation(offerHandle, brandKeywordRecord),
-      getCurrentAllocations: (offerHandles, brandKeywordRecords) =>
-        doGetCurrentAllocations(offerHandles, brandKeywordRecords),
-      getInstallation: installationHandle =>
-        installationTable.get(installationHandle).bundle,
+    isOfferActive: offerTable.isOfferActive,
+    getOffers: offerHandles =>
+      offerTable.getOffers(offerHandles).map(removeAmountsAndNotifier),
+    getOffer: offerHandle =>
+      removeAmountsAndNotifier(offerTable.get(offerHandle)),
+    getCurrentAllocation: (offerHandle, brandKeywordRecord) =>
+      doGetCurrentAllocation(offerHandle, brandKeywordRecord),
+    getCurrentAllocations: (offerHandles, brandKeywordRecords) =>
+      doGetCurrentAllocations(offerHandles, brandKeywordRecords),
+    getInstallation: installationHandle =>
+      installationTable.get(installationHandle).bundle,
+  });
+
+  return harden({
+    buildZoeService: vatAdminWrapper => {
```

Comment:
Can you explain more about why there is a `makeZoe` call and a `buildZoeService` call? It seems like we could just have a single `makeZoe` call

Reply:
You're absolutely right. I think Brian pointed this out earlier, and I missed it in my clean up pass.  I will make that go away.